### ISSUE TITLE
[Feature Request] mTLS sample

### ISSUE THREAD

**Issue by bergundy** [temporalio/samples-java#331]
This sample exists but we need one for mtls for both worker and "starter".

https://github.com/temporalio/samples-java/tree/main/src/main/java/io/temporal/samples/ssl

The sample should be modeled after https://github.com/temporalio/samples-typescript/tree/main/hello-world-mtls

**Comment by bergundy** [comment]
Break this sample out to separate files to align with the other SDK sample repos

**Comment by cretz** [comment]
> Break this sample out to separate files to align with the other SDK sample repos

This repo has many single-file examples, so this aligns with itself. Unless we're considering breaking all of them out to align with other SDK sample repos, I am not sure just this sample deserves to be special (though it's harmless, but using other SDK samples repos as justification to require this be broken out but none of the others is inconsistent).

**Comment by bergundy** [comment]
Let me rephrase the request, we need to embed the snippets from the samples repo into the docs, if a single file sample works here that should be fine.
